### PR TITLE
feat(css): Add example of `@counter-style` and all of its descriptors

### PR DESCRIPTION
**Pull request by NiedziolkaMichal**
It adds interactive examples to @counter-style and all its descriptors except for [speak-as](https://developer.mozilla.org/en-US/docs/Web/CSS/@counter-style/speak-as), which doesn't have any visual impact.

image

image

image

image

image

image

image

image

image

image

### CODE REVIEW

**Comment by wbamberg** [comment]
Thanks for this PR, @NiedziolkaMichal ! I don't have time to review now, but:

>     2. They don't work on firefox. It looks like latest version of firefox has an issue, with @counter-style and possibly other at-rules, be not operational, while being placed in shadow-root.

I remembered this after our conversation: https://github.com/mdn/interactive-examples/issues/887 . Maybe the same thing?

**Comment by NiedziolkaMichal** [comment]
Yes, it looks to be the same issue, although @font-face doesn't work on both firefox & chrome, while @counter-style works fine on chromium based browsers. Is there any way to make it work?

**Comment by wbamberg** [comment]
I don't know, @NiedziolkaMichal , sorry, @schalkneethling , do you have any ideas about this? I'm afraid that this is quite likely a blocker for this idea.

There is more information here:

* https://github.com/w3c/csswg-drafts/issues/5693
* https://bugzilla.mozilla.org/show_bug.cgi?id=1672302
* https://wiki.csswg.org/spec/css-scoping

**Comment by schalkneethling** [comment]
@NiedziolkaMichal thank you for your continued contributions. I believe your recent pull request on mdn/bob will address the issues mentioned here? If so, let’s keep this one around until that one is merged. Thanks!

**Comment by NiedziolkaMichal** [comment]
@schalkneethling No, this PR will still not work on Firefox, because output is presented in shadow tree. Recently I have been prototyping on replacing shadow DOM with iframe in HTML interactive examples. It would make it possible to use all at-rules and WebAPI examples, without much effort. Is there any important reason why you have chosen shadow dom instead?

**Comment by schalkneethling** [comment]
> Is there any important reason why you have chosen shadow dom instead?

I honestly cannot remember :) I believe it could have been because we were, and still are, considering moving interactive examples into content without it being hosted in an `iframe` at all. We needed to prevent styling and Javascript "leaking" into the rest of the page.

**Comment by NiedziolkaMichal** [comment]
@schalkneethling Making HTML examples be operational without either iframe or shadow DOM would be tricky. There are plenty of examples, like [p](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/p), using selectors that would affect rest of the layout.

**Comment by wbamberg** [comment]
There are I think 2 levels here:
- the complete example, including the UI, code editors, and output
- the output pane

At the moment the complete example is in an iframe, of course, and the output pane is in shadow DOM to stop it leaking into the editor UI. So even if the complete example was "inline" in the page, we'd need to isolate the output.

> Recently I have been prototyping on replacing shadow DOM with iframe in HTML interactive examples.

That's interesting! So we'd have an iframe inside an iframe? I might have mentioned this a few thousand times before, but I'd love to have Web API interactive examples on MDN.

**Comment by wbamberg** [comment]
> @schalkneethling No, this PR will still not work on Firefox, because output is presented in shadow tree. Recently I have been prototyping on replacing shadow DOM with iframe in HTML interactive examples. It would make it possible to use all at-rules and WebAPI examples, without much effort. Is there any important reason why you have chosen shadow dom instead?

Hey, @NiedziolkaMichal , I have been trying to write this whole thing up and found this: https://github.com/mdn/interactive-examples/issues/706 which is where we decided to use shadow DOM. It was also suggested there that an extra iframe could be a solution.

Just in case this bit of history could help :).

**Comment by NiedziolkaMichal** [comment]
@wbamberg In the [iframe PR](https://github.com/mdn/bob/pull/779), I believe that those issues don't exist. Applied CSS rules, don't select anything other than output. Clicking on links opens new browser tab. My solution also don't generate two separate documents, output exists only in memory and is applied via srcdoc attribute. On my device, I also didn't notice any performance issues.

File: live-examples/css-examples/@counter-style/counter-style.css [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@counter-style black-white-squares {
+    system: symbolic;
+    symbols: ◆ ◇;
+    suffix: ' ';
```
Comment (estelle) [inline live-examples/css-examples/@counter-style/counter-style.css]:
```suggestion
    suffix: " ";
```

File: live-examples/css-examples/@counter-style/negative.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@counter-style negative-alphabet {
+    system: numeric;
+    symbols: 'A' 'B' 'C' 'D' 'E' 'F';
+    negative: '→' '←';
```
Comment (estelle) [inline live-examples/css-examples/@counter-style/negative.css]:
```suggestion
    symbols: "A" "B" "C" "D" "E" "F";
    negative: "→" "←";
```

File: live-examples/css-examples/@counter-style/pad.css [hunk 3]
```diff
@@ -0,0 +1,9 @@
+@counter-style binary-with-pad {
+    system: numeric;
+    symbols: '0' '1';
+    pad: 4 '_';
```
Comment (estelle) [inline live-examples/css-examples/@counter-style/pad.css]:
```suggestion
    symbols: "0" "1";
    pad: 4 "_";
```

File: live-examples/css-examples/@counter-style/prefix.css [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@counter-style heads {
+    system: cyclic;
+    symbols: 👧 🧒 🧑 👱;
+    prefix: '🖐🏻';
```
Comment (estelle) [inline live-examples/css-examples/@counter-style/prefix.css]:
```suggestion
    symbols: 👧 🧒 🧑 👱;
    prefix: "🖐🏻";
```

File: live-examples/css-examples/@counter-style/range.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
+@counter-style missing-shapes {
+    system: cyclic;
+    symbols: '♥︎' '♤';
```
Comment (estelle) [inline live-examples/css-examples/@counter-style/range.css]:
```suggestion
    symbols: "♥︎" "♤";
```

File: live-examples/css-examples/@counter-style/suffix.css [hunk 6]
```diff
@@ -0,0 +1,9 @@
+@counter-style octal {
+    system: numeric;
+    symbols: '0' '1' '2' '3' '4' '5' '6' '7';
+    suffix: ': ';
```
Comment (estelle) [inline live-examples/css-examples/@counter-style/suffix.css]:
```suggestion
    symbols: "0" "1" "2" "3" "4" "5" "6" "7";
    suffix: ": ";
```

File: live-examples/css-examples/@counter-style/symbols.css [hunk 7]
```diff
@@ -0,0 +1,8 @@
+@counter-style cool-symbols {
+    system: symbolic;
+    symbols: '*' '\21C9'▲ 🐉 '🥲' ✹ ❎ 💓;
```
Comment (estelle) [inline live-examples/css-examples/@counter-style/symbols.css]:
```suggestion
    symbols: "*" "\21C9" ▲ 🐉 "🥲" ✹ '"' ❎ 💓;
```
i added a double quote here to show that quotes can be included

**Review by estelle — COMMENTED** [review]
Looks good. My suggestions include adding white space, using double quotes, and adding an example of a using a quote as a symbol. 

I created a PR to add a comment about white space and quotes; so our example follow our best practices, but the content indicates that if you don't it should still work. 

https://github.com/mdn/content/pull/26677

**Comment by bsmth** [comment]
> My suggestions include adding white space, using double quotes

The formatter doesn't seem to like this, so I've added in https://github.com/mdn/interactive-examples/pull/2089/commits/f4c78a49bca1949f36a2bd8a0dc58b0a2f7464b9 to make things green, I hope that's okay.

File: live-examples/css-examples/@counter-style/additive-symbols.css [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    400 cd,
+    100 c,
+    90 xc,
+    50 l,
+    40 xl,
+    10 x,
+    9 ix,
+    5 v,
+    4 iv,
+    1 i;
```
Comment (bsmth) [inline live-examples/css-examples/@counter-style/additive-symbols.css]:
```suggestion
  additive-symbols: 50 l, 40 xl, 10 x, 9 ix, 5 v, 4 iv, 1 i;
```

**Review by bsmth — APPROVED** [review]
One suggestion open to prevent the line wrapping, but looks good aside from that, leaving a +1 👍🏻

**Comment by bsmth** [comment]
Thanks everyone for the work! Merging shortly 🚢

**Pull request merged**